odoo/addons/base/models/ir_async.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+import itertools
+import json
+import logging
+import re
+import traceback
+import threading
+import odoo
+import odoo.exceptions
+from odoo import models, fields, api
+from odoo.http import serialize_exception
+
+_logger = logging.getLogger(__name__)
+BASE_VERSION = odoo.modules.load_information_from_description_file('base')['version']
+
+
+class IrAsync(models.Model):
+    _name = 'ir.async'
+    _description = 'Asynchronous Jobs'
+
+    name = fields.Char()
+    state = fields.Selection([
+        ('created', 'Created'),     # The job has been enqueued for later processing
+        ('succeeded', 'Succeeded'), # The job finished with a result, the result has not yet been retrieved
+        ('failed', 'Failed'),       # The job failed due to an exception
+        ('done', 'Done'),           # The job finished without result or the result has been retrieved
+    ])
```

Comment:
+ default = "created" ?

Reply:
actually only sudo can create a new ir.async and the fields are all set

Reply:
so I don't know if it would be better to set all the fields as required ?

Reply:
I just find it nice when the SQL models is strictly specified from the beginning to avoid potential inconsistencies ;).  You never know how your feature will be used (even if indeed, they should use your API and avoid manual creation of ir_async records).